storage/memory/peer_store.go

```diff
@@ -125,6 +125,9 @@ func New(cfg Config) (storage.PeerStore, error) {
 	ps.wg.Add(1)
 	go func() {
 		defer ps.wg.Done()
+		if cfg.PrometheusReportingInterval == 0 {
```

Comment:
Make this check for `<=0`. `time.Duration` is actually an `int64`, so someone could set this to a negative value.